Initialize App data from Firestore

jpoist97 opened the pull request:
Redux is now initialized from Firestore. We need to do a bit more work on the DraggableWorkoutList since it was displaying the exerciseIds and not the names. Also I left color out of Firestore for now since we haven't decided what we want to do for that. But I'll add that in a later PR if that's what we decide to go with.

jpoist97 committed:
Initialize App data from firestore

Redux is now initialized from firestore. We need to do a bit more work on the DraggableWorkoutList since it was displaying the exerciseIds and not the names.

Also I left color out of Firstore for now since we haven't decided what we want to do for that. But I'll add that in a later PR if that's what we decide to go with.

finlaylp reviewed (dismissed):
Looks good. So this works assuming the data has already been initialized for the user right? Which is the task I'm assigned

jpoist97 commented:
> for the user right? Which is the task I'm assigned

Yeah, I guess we should either hold off on merging this PR in, or we create like a test account with some test data so when we're developing the app will be filled out still.

jpoist97 committed:
Add error protections

Added some error protections. With the new updated workout structure, addWorkout needs to be updated slightly to create a properly shaped workout object

jpoist97 commented on components/Home/Home.jsx:
```diff
@@ -27,17 +29,87 @@ const welcomeName = 'Shriya';
-    dispatch({ type: INITIALIZE_WORKOUTS, workouts: DBWorkoutResponse });
+    const initializeDatabase = async () => {
+      // Get the current logged in user id
+      const currentUser = firebase.auth().currentUser.uid;
+      // const currentUser = '68w6wWz8l5QJO3tDukh1fRXWYjD2';
```
Forgot to mention this, but when you're running this locally before Fin's task to update firebase when a user is created, comment out line 34 and uncomment line 35 and you'll be able to get the test data in my personal account

If you don't do this the app will error out bc there will be nothing in it

ctperry0301 commented on components/LogWorkout/LogWorkout.jsx:
```diff
@@ -23,18 +23,22 @@ const TitleText = styled.Text`
+    sets: exercise.sets.map((set) => ({
+      prevWeight: set.weight && set.weight.toString(),
+      weight: '',
+      prevDuration: exerciseType === 'REPS' ? set.reps.toString() : set.time.toString(),
```
The set.reps.toString() and set.time.toString() will cause an error if they are left blank

jpoist97 replied:
Yeah, they should never both be undefined though. They will be for newly created workouts but once AddWorkout is updated to reflect the new workout structure it should never occur

ctperry0301 commented on components/CreateCycle/ExpandableWorkoutCard.jsx:
```diff
@@ -76,12 +76,12 @@ const ExpandableWorkoutCard = (props) => {
         <FlatList
           data={exercises}
           // TODO: rename to exercise name
```
can these TODO's be completed now?

jpoist97 replied:
Yeah, I'll take care of it

ctperry0301 reviewed (commented):
Yeah, we will definitely need a loading page. 
See comments, close to approval

jpoist97 committed:
Update ExpandableWorkoutCard to use exercise names

Updated ExpandableWorkoutCard to use actual Exercise names instead of ids

jpoist97 commented on components/CreateCycle/ExpandableWorkoutCard.jsx:
```diff
@@ -74,14 +89,13 @@ const ExpandableWorkoutCard = (props) => {
             <RowContent>
-              <WorkoutViewText style={{ marginLeft: 10 }}>{item.id}</WorkoutViewText>
+              <WorkoutViewText style={{ marginLeft: 10 }}>{item.name}</WorkoutViewText>
               {/* TODO: center the sets x reps around the x so double digit numbers dont look bad */}
```
Aye @finlaylp do you remember what this todo was referring to?

finlaylp replied:
yeah that was so that like 3 x 12 and 3 x 8 would be aligned around the x instead of just centered, bc it looks a bit odd that the x's can sometimes be offset a bit. Not a big issue but I didn't get around to it.

jpoist97 committed:
Optimize Firestore retrieval

Retrieve everything from Firebase in parrallel

ctperry0301 reviewed (approved):
Loads much faster now, Looks good.
